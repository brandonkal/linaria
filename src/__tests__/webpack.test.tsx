import compiler from './__utils__/compiler';
import rimraf from 'rimraf';

describe('webpack test', () => {
  beforeEach(() => {
    rimraf.sync('.linaria-cache');
  });

  test(
    'project compiles -- optimizer',
    async () => {
      const FILE = 'project.tsx';
      const stats = await compiler({
        fixture: FILE,
        folder: 'project',
        production: true,
        complex: true,
        optimize: true,
      });
      const modules: any[] = stats.toJson().modules;
      let js = modules.find(v => v.name.includes(FILE));
      if (js.modules && js.modules.length) {
        js = js.modules.find(v => v.name.includes(FILE));
      }
      const jsSrc = js.source;
      const bundle: string = stats.compilation.assets['bundle.js'].source();
      const css: string = stats.compilation.assets['styles.css'].source();
      expect(css.includes('LINARIA')).toBe(false);
      expect(bundle.includes('LINARIA')).toBe(false);
      const snap =
        '// JS after Babel plugin before optimization\n' +
        jsSrc +
        '\n\n/* CSS after optimization */\n' +
        css;

      expect(snap).toMatchSnapshot();
    },
    3600 * 1000
  );
});
