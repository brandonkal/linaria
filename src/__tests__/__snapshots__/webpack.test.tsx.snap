// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`webpack test project compiles -- optimizer 1`] = `
"// JS after Babel plugin before optimization
import React from 'react'; //@ts-ignore

import { styled } from '@brandonkal/linaria/react';
import render from 'react-test-renderer'; // @ts-ignore -- uses webpack alias

import { Alert } from 'components';
var Container =
/*#__PURE__*/
styled(\\"div\\")({
  name: \\"Container\\",
  class: \\"LINARIA_c1ckeoc9_LINARIA\\"
});
var toLog = 'hello';
render.create(React.createElement(Container, null, React.createElement(Alert, {
  variant$: \\"error\\",
  title$: \\"Error\\",
  onClose: () => console.log('You closed the message')
}), React.createElement(Alert, {
  variant$: \\"error\\",
  title$: \\"Error\\"
}))); // Injected by Linaria Loader for Webpack Compilation (virtual)

require(\\"./.linaria-cache/css/7092bd38278643717e46e36be1c3e8d5.prod.linaria.css\\");

/* CSS after optimization */

















.a {
  top: 15px;
  color: rgba(0, 0, 0, 0.45);
  cursor: pointer;
}
  .a:hover {
    color: rgba(0, 0, 0, 0.75);
  }
.c {
  font-size: 14px;
}
.d {
  color: red;
  font-weight: 700;
}










.e {
  position: relative;
  margin: 0;
  padding: var(--e-0);
  color: rgba(0, 0, 0, 0.65);
  font-size: 14px;
}










.e.e-1 {
    margin-bottom: 0;
    border: 0;
  }










.e.e-2 {
    background-color: #f6ffed;
    border: 1px solid #b7eb8f;
  }







.b {
  margin-bottom: 10px;
}
  .b e {
    background: pink;
    padding: 1rem;
  }
"
`;
